service/api_task.go

```diff
@@ -223,7 +228,11 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// mark task as finished
-	err = model.MarkEnd(t, APIServerLockTitle, finishTime, details, projectRef.ShouldDeactivatePrevious())
+	deactivatePrevious := projectRef.ShouldDeactivatePrevious()
+	if utility.FromBoolPtr(projectParser.DeactivatePrevious) {
```

Comment:
I might have a preference for making a ShouldDeactivatePrevious function for project parser to be consistent. This was meant to make the code a bit more readable.